torch/distributed/nn/functional.py

```diff
@@ -207,6 +232,20 @@ def backward(ctx, grad_output):
         return (None, None, None) + (_Broadcast.apply(ctx.src, ctx.group, grad_output),)
 
 
+class _Reduce_Scatter(Function):
+    @staticmethod
+    def forward(ctx, op, group, tensor, *input_tensor_list):
+        ctx.group = group
+        dist.reduce_scatter(tensor, list(input_tensor_list), op=op, group=group)
+        return tensor
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        result = _AllGather.apply(ctx.group, grad_output.contiguous())
+        print("backward ", grad_output, result)
+        return (None, None, None) + _AllGather.apply(ctx.group, grad_output.contiguous())
```

Comment:
Shouldn't we return `result` here? Why are we calling allgather again?

Reply:
opps... Sorry I forgot removing the debugging code...